x-pack/plugin/esql/qa/testFixtures/src/main/resources/union_types.csv-spec

```diff
@@ -0,0 +1,644 @@
+singleIndexIp
+FROM sample_data
+| EVAL client_ip = TO_IP(client_ip)
+| KEEP @timestamp, client_ip, event_duration, message
+| SORT @timestamp DESC
+;
+
+@timestamp:date           |  client_ip:ip  |  event_duration:long  |  message:keyword
+2023-10-23T13:55:01.543Z  |  172.21.3.15   |  1756467              |  Connected to 10.1.0.1
+2023-10-23T13:53:55.832Z  |  172.21.3.15   |  5033755              |  Connection error
+2023-10-23T13:52:55.015Z  |  172.21.3.15   |  8268153              |  Connection error
+2023-10-23T13:51:54.732Z  |  172.21.3.15   |  725448               |  Connection error
+2023-10-23T13:33:34.937Z  |  172.21.0.5    |  1232382              |  Disconnected
+2023-10-23T12:27:28.948Z  |  172.21.2.113  |  2764889              |  Connected to 10.1.0.2
+2023-10-23T12:15:03.360Z  |  172.21.2.162  |  3450233              |  Connected to 10.1.0.3
+;
+
+singleIndexWhereIpLike
+FROM sample_data
+| WHERE TO_STRING(client_ip) LIKE "172.21.2.*"
+| KEEP @timestamp, event_duration, message
+| SORT @timestamp DESC
+;
+
+@timestamp:date           |  event_duration:long  |  message:keyword
+2023-10-23T12:27:28.948Z  |  2764889              |  Connected to 10.1.0.2
+2023-10-23T12:15:03.360Z  |  3450233              |  Connected to 10.1.0.3
+;
+
+singleIndexTsLong
+FROM sample_data_ts_long
+| EVAL @timestamp = TO_DATETIME(@timestamp)
+| KEEP @timestamp, client_ip, event_duration, message
+| SORT @timestamp DESC
+;
+
+@timestamp:date           |  client_ip:ip  |  event_duration:long  |  message:keyword
+2023-10-23T13:55:01.543Z  |  172.21.3.15   |  1756467              |  Connected to 10.1.0.1
+2023-10-23T13:53:55.832Z  |  172.21.3.15   |  5033755              |  Connection error
+2023-10-23T13:52:55.015Z  |  172.21.3.15   |  8268153              |  Connection error
+2023-10-23T13:51:54.732Z  |  172.21.3.15   |  725448               |  Connection error
+2023-10-23T13:33:34.937Z  |  172.21.0.5    |  1232382              |  Disconnected
+2023-10-23T12:27:28.948Z  |  172.21.2.113  |  2764889              |  Connected to 10.1.0.2
+2023-10-23T12:15:03.360Z  |  172.21.2.162  |  3450233              |  Connected to 10.1.0.3
+;
+
+singleIndexIpStats
+FROM sample_data
+| EVAL client_ip = TO_IP(client_ip)
+| STATS count=count(*) BY client_ip
+| SORT count DESC, client_ip ASC
+| KEEP count, client_ip
+;
+
+count:long  |  client_ip:ip  
+4           |  172.21.3.15
+1           |  172.21.0.5
+1           |  172.21.2.113
+1           |  172.21.2.162
+;
+
+singleIndexIpStringStats
+FROM sample_data_str
+| EVAL client_ip = TO_IP(client_ip)
+| STATS count=count(*) BY client_ip
```

Comment:
Please add a variant with the shorthand example:
```
from sample
| stats count = count(*) by client_ip = to_ip(client_ip)
...
```
And another one with the conversion inside the aggregation which later gets dropped:

```
from index
| stats m = max(to_string(client_ip)), c = count()
| keep c
```